### PR TITLE
PP-576 Refactor connectorApp

### PR DESCRIPTION
**Pull request by rauligs**
- As part of the migration to JPA, Guice was introduced for dependency injection so current explicitly declared dependencies are now automatically instrumented by Guice. 
- ConnectorModule is the sole responsible to manage binding (JpaPersistenceModule installed within the Connector Module).

### CODE REVIEW

File: src/main/java/uk/gov/pay/connector/app/ConnectorModule.java [hunk 1]
```diff
@@ -19,6 +25,30 @@ public ConnectorModule(final ConnectorConfiguration configuration, final Environ
...
         bind(ConnectorConfiguration.class).toInstance(configuration);
         bind(Environment.class).toInstance(environment);
+
         bind(ChargeEventJpaListener.class).in(Singleton.class);
+        bind(ITokenDao.class).to(TokenJpaDao.class);
+        bind(IChargeDao.class).to(ChargeJpaDao.class);
+        bind(IEventDao.class).to(EventJpaDao.class);
+        bind(IGatewayAccountDao.class).to(GatewayAccountJpaDao.class);
+
+        install(jpaModule(configuration.getDataSourceFactory()));
```
Comment (kakumara) [inline src/main/java/uk/gov/pay/connector/app/ConnectorModule.java]: Nice :+1:

**Comment by ijmad** [comment]
Could we put a better description on the PR as well as the commits please?
Why are we refactoring, what are we refactoring, etc.

Thanks!

**Comment by rauligs** [comment]
Added description

File: src/main/java/uk/gov/pay/connector/auth/SmartpayAuthenticator.java [hunk 2]
```diff
@@ -4,6 +4,7 @@
 import io.dropwizard.auth.AuthenticationException;
 import io.dropwizard.auth.Authenticator;
 import io.dropwizard.auth.basic.BasicCredentials;
+import uk.gov.pay.connector.app.ConnectorConfiguration;
```
Comment (kakumara) [inline src/main/java/uk/gov/pay/connector/auth/SmartpayAuthenticator.java]: remove unused import

File: src/main/java/uk/gov/pay/connector/resources/ChargesApiResource.java [hunk 3]
```diff
@@ -73,12 +75,13 @@
     private static final Logger logger = LoggerFactory.getLogger(ChargesApiResource.class);
     private static final DateFormat DATE_FORMAT = new SimpleDateFormat("dd/MM/yyyy HH:mm:ss");
 
-    public ChargesApiResource(IChargeDao chargeDao, ITokenDao tokenDao, IGatewayAccountDao gatewayAccountDao, IEventDao eventDao, LinksConfig linksConfig) {
+    @Inject
+    public ChargesApiResource(IChargeDao chargeDao, ITokenDao tokenDao, IGatewayAccountDao gatewayAccountDao, IEventDao eventDao, ConnectorConfiguration configuration) {
```
Comment (rahul-vinayak) [inline src/main/java/uk/gov/pay/connector/resources/ChargesApiResource.java]: Was part of the refactoring exercise to get rid of the interfaces??

**Pull request merged**